libtransmission/crypto-utils-cyassl.cc

```diff
@@ -139,20 +139,17 @@ bool tr_sha1_update(tr_sha1_ctx_t raw_handle, void const* data, size_t data_leng
     return check_result(API(ShaUpdate)(handle, static_cast<byte const*>(data), data_length));
 }
 
-bool tr_sha1_final(tr_sha1_ctx_t raw_handle, uint8_t* hash)
+std::optional<tr_sha1_digest_t> tr_sha1_final(tr_sha1_ctx_t raw_handle)
 {
     auto* handle = static_cast<Sha*>(raw_handle);
-    bool ret = true;
-
-    if (hash != nullptr)
-    {
-        TR_ASSERT(handle != nullptr);
-
-        ret = check_result(API(ShaFinal)(handle, hash));
-    }
+    TR_ASSERT(handle != nullptr);
 
+    auto digest = tr_sha1_digest_t{};
+    auto digest_in_cyassl = reinterpret_cast<unsigned char*>(std::data(digest));
+    auto const ok = check_result(API(ShaFinal)(handle, digest_in_cyassl));
     tr_free(handle);
-    return ret;
+
+    return ok ? digest : std::optional<tr_sha1_digest_t>{};
```

Comment:
```suggestion
    return ok ? digest : std::nullopt;
```

Reply:
When I make this change in a similar section in crypto-utils-openssl, I get this:

```
crypto-utils-openssl.cc:145:15: error: incompatible operand types ('std::array<std::byte, 20>' and 'const std::nullopt_t') [clang-diagnostic-error]
    return ok ? digest : std::nullopt;
              ^ ~~~~~~   ~~~~~~~~~~~~
```

Reply:
Ah nvm, `digest` needs to be replaced with `std::make_optional()` to make the type match with `std::nullopt`. Fixed in 92b2d938496e0e0107cd8ecd98dd6a07c6d90b1f